communication/src/spark_protocol.cpp

```diff
@@ -1198,6 +1200,33 @@ void SparkProtocol::set_chunks_received(uint8_t value)
     	memset(queue+QUEUE_SIZE-bytes, value, bytes);
 }
 
+size_t SparkProtocol::notify_update_done(uint8_t* msg, token_t token, uint8_t code)
+{
+    size_t msgsz = 0;
+    char buf[255];
+    size_t data_len = 0;
+
+    memset(buf, 0, sizeof(buf));
+    if (code != ChunkReceivedCode::BAD) {
+        callbacks.finish_firmware_update(file, UPDATE_FLAG_SUCCESS | UPDATE_FLAG_VALIDATE_ONLY, buf);
+        data_len = strnlen(buf, sizeof(buf) - 1);
```

Comment:
Do we include the length of the extra payload info in `buf`? If not, do we just rely on the `size_t` of the packet to know where the payload ends, since we are not sending the null terminator `sizeof(buf) - 1`.